apps/native/app/src/stores/environment-store.ts

```diff
@@ -73,10 +74,17 @@ export const environmentStore = create<EnvironmentStore>(
                 'X-Cognito-Token': `Bearer ${get().cognito?.accessToken}`,
               },
             }).then((r) => r.json() as Promise<EnvironmentResponse>)
-            const ids = res.results.ids.map((n) => ({
-              ...((environments as any)[n.id] ?? {}),
-              ...n,
-            }))
+            const ids = res.results.ids.map((n) => {
+              const local =
+                environments[n.id as keyof typeof environments] ?? {}
+              const remote = n as any
+              return {
+                ...local,
+                ...n,
+                // Preserve baseUrl from local if remote does not provide it
+                baseUrl: remote.baseUrl ?? local.baseUrl,
```

Comment:
Should maybe guess it from the apiUrl instead to better support feature deployments?